src/frida/repl.py

```diff
@@ -51,6 +56,17 @@ def _start(self):
                 self._update_status("Failed to load script: {error}".format(error=e))
                 self._exit(1)
                 return
+
+            if self._watch:
+                handler_dir = os.path.dirname(self._user_script)
+                monitor = self._script_monitor
+                if monitor is None:
+                    monitor = frida.FileMonitor(handler_dir)
```

Comment:
We shouldn't be monitoring the directory here, it's better to just monitor the file itself.